### PR TITLE
[SIMPLE_FORMS] feat: add logic to handle multiple attachments for 20-10207

### PR DESCRIPTION
**Pull request by pennja**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES/NO*
- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by coope93 — APPROVED** [review]
Nice job thinning out the controller 👍

File: modules/simple_forms_api/app/services/simple_forms_api/benefits_intake_client.rb [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      '21-4142' => 'vba_21_4142',
+      '21P-0847' => 'vba_21p_0847',
+      '26-4555' => 'vba_26_4555',
+      '40-0247' => 'vba_40_0247',
+      '40-10007' => 'vba_40_10007'
+    }.freeze
+
+    def initialize(current_user, form_number:, **kwargs)
+      @current_user = current_user
+      @form_number = form_number
```
Comment (Thrillberg) [inline modules/simple_forms_api/app/services/simple_forms_api/benefits_intake_client.rb:22]: Do we ever use the `form_number` argument in initialization?

File: modules/simple_forms_api/app/services/simple_forms_api/benefits_intake_client.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        @attachments = form.get_attachments
+      elsif %w[vba_40_0247 vba_40_10007].include?(form_id)
+        form.handle_attachments(file_path)
+      end
+    end
+
+    def upload_pdf
+      location, uuid = prepare_for_upload
+      log_upload_details(location, uuid)
+      response = use_benefits_intake_service? ? perform_pdf_upload(location) : perform_document_upload(location)
```
Comment (Thrillberg) [inline modules/simple_forms_api/app/services/simple_forms_api/benefits_intake_client.rb:128]: I think we actually want to use [`SimpleFormsApi::PdfUploader`](https://github.com/department-of-veterans-affairs/vets-api/blob/b760b06482a687675d24e84e6797a1d47a661d16/modules/simple_forms_api/app/controllers/simple_forms_api/v1/uploads_controller.rb#L129) here, if the flipper is disabled.

**Review by Thrillberg — COMMENTED** [review]
Looking good to me! I have two comments inside plus I'm wondering if we can write or update a test to ensure that we're using the `attachments` argument for `BenefitsIntake::Service.perform_upload`.

File: modules/simple_forms_api/app/services/simple_forms_api/benefits_intake_client.rb (hunk 1 above)
Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/benefits_intake_client.rb:22]: Hmm, good point. I'm just assigning the global variable here, I could always do `@form_number = kwargs[:form_number]` instead.

**Comment by stevenjcumming** [comment]
@pennja is this still being worked on or can this PR be closed?

**Comment by stevenjcumming** [comment]
@pennja is this good to close?

**Comment by pennja** [comment]
> @pennja is this good to close?

I'd like to keep the refactor work here but the PR itself is not longer necessary. I'll make a fresh one based off this one.

**Pull request closed without merging**